master/buildbot/data/resultspec.py

```diff
@@ -109,6 +109,14 @@ def nonecmp(a, b):
     return cmp(a, b)
 
 
+class MinType(object):
```

Comment:
This is a clever trick, but needs at least a comment on why it does this like that.